src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/TypeMustExistTests.cs

```diff
@@ -276,5 +279,271 @@ internal class InternalNested { }
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+          public string MyField;
+        }
+      }
+    }
+  }
+}
+";
+
+            string[] rightSyntaxes = new[]
+            { @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+      }
+    }
+  }
+}
+",
+            @"
+namespace CompatTests
+{
+  public class First
+  {
+  }
+}
+",
+            @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+    }
+  }
+}
+",
+            @"
+namespace CompatTests
+{
+}
+"};
+
+            ApiComparer differ = new();
+            ElementContainer<IAssemblySymbol> left =
+                new(SymbolFactory.GetAssemblyFromSyntax(leftSyntax), new MetadataInformation(string.Empty, string.Empty, "ref"));
+
+            IList<ElementContainer<IAssemblySymbol>> right = SymbolFactory.GetElementContainersFromSyntaxes(rightSyntaxes);
+
+            IEnumerable<(MetadataInformation, MetadataInformation, IEnumerable<CompatDifference>)> differences =
+                differ.GetDifferences(left, right);
+
+            List<(MetadataInformation, MetadataInformation, IEnumerable<CompatDifference>)> expected = new()
+            {
+                (left.MetadataInformation, new MetadataInformation(string.Empty, string.Empty, "runtime-0"), new CompatDifference[]
+                {
+                    new CompatDifference(DiagnosticIds.TypeMustExist, $"Type 'CompatTests.First.FirstNested.SecondNested.ThirdNested' exists on the left but not on the right", DifferenceType.Removed, "T:CompatTests.First.FirstNested.SecondNested.ThirdNested"),
+                }),
+                (left.MetadataInformation, new MetadataInformation(string.Empty, string.Empty, "runtime-1"), new CompatDifference[]
+                {
+                    new CompatDifference(DiagnosticIds.TypeMustExist, $"Type 'CompatTests.First.FirstNested' exists on the left but not on the right", DifferenceType.Removed, "T:CompatTests.First.FirstNested"),
+                }),
+                (left.MetadataInformation, new MetadataInformation(string.Empty, string.Empty, "runtime-2"), new CompatDifference[]
+                {
+                    new CompatDifference(DiagnosticIds.TypeMustExist, $"Type 'CompatTests.First.FirstNested.SecondNested' exists on the left but not on the right", DifferenceType.Removed, "T:CompatTests.First.FirstNested.SecondNested"),
+                }),
+                (left.MetadataInformation, new MetadataInformation(string.Empty, string.Empty, "runtime-3"), new CompatDifference[]
+                {
+                    new CompatDifference(DiagnosticIds.TypeMustExist, $"Type 'CompatTests.First' exists on the left but not on the right", DifferenceType.Removed, "T:CompatTests.First"),
+                }),
```

Comment:
I think we should develop better design model so that we can add new tests relatively simpler. Not a blocker for pr or for preview, just something to think about

Reply:
I think the way I'm doing this expected vs actual can definitely be simplified. I'll look into simplifying it for the single left vs multiple left scenarios.

Reply:
Discussed offline.